test/migrate/fixtures/legacy-for-syntax/snapshot-expected.marko

```diff
@@ -0,0 +1,141 @@
+<!-- test/migrate/fixtures/legacy-for-syntax/template.marko -->
+
+<!-- Array -->
+<for|color| of=colors>
+    <li>${color}</li>
+</for>
+<!-- Array | iterator -->
+$ var iteratorColors = [];
+$ input.iterator(colors, function(item) {
+    iteratorColors.push(item);
+});
+<for|color| of=iteratorColors>
+    <li>${color}</li>
+</for>
+<!-- Array | separator -->
+<for|color, loopIndex, loopAll| of=colors>
+    <li>${color}</li>
+    <if(loopIndex !== loopAll.length - 1)>,</if>
+</for>
+<!-- Array | status-var -->
+<for|color, loopIndex, loopAll| of=colors>
+    ${color}${loopIndex + 1}) of ${loopAll.length}<if(loopIndex === 0)>
+        - FIRST
+    </if>
+    <if(loopIndex === loopAll.length - 1)>- LAST</if>
+</for>
+<!-- Array | iterator separator -->
+$ var iteratorColors = [];
+$ input.iterator(colors, function(item) {
+    iteratorColors.push(item);
+});
+<for|color, loopIndex, loopAll| of=iteratorColors>
+    <li>${color}</li>
+    <if(loopIndex !== loopAll.length - 1)>,</if>
+</for>
+<!-- Array | status-var iterator -->
+<for|color, loop| of=input.iterator.bind(null, colors)>
+    ${color}${loop.getIndex() + 1}) of ${loop.getLength()}<if(loop.isFirst())>
```

Comment:
Unmigrated:
```
${loop.getIndex() + 1}) of ${loop.getLength()}
```

Reply:
This is intentional when you have both `status-var` and `iterator`. In the current for loop the custom iterator is responsible for providing the `status-var`.